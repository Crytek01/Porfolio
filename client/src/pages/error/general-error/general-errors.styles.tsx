import styled from 'styled-components';

import { ReactComponent as ErrorSVG } from '../../../assets/icons/plugs-connection-outline-svg.svg';

export const GeneralErrorContainer = styled.div`
  padding-top: 2rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const ErrorImage = styled(ErrorSVG)`
  display: block;
  width: clamp(15rem, 100%, 25rem);
  height: clamp(15rem, 100%, 25rem);

  fill: #333;
`;

export const Title = styled.h1`
  font-size: ${({ theme: { fontSize } }) => fontSize[8]};
  font-weight: ${({ theme: { fontWeight } }) => fontWeight.semibold};
  line-height: 1.2;

  margin-bottom: ${({ theme: { spacing } }) => spacing[9]};
`;

export const Subtitle = styled.p`
  font-size: ${({ theme: { fontSize } }) => fontSize[5]};
  line-height: 1.2;
  margin-bottom: 16rem;
  color: #777;
`;
