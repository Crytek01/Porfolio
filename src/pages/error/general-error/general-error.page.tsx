import { FunctionComponent } from 'react';

import {
  ErrorImage,
  GeneralErrorContainer,
  Subtitle,
  Title,
} from './general-errors.styles';

export const GeneralErrorPage: FunctionComponent = () => (
  <GeneralErrorContainer>
    <ErrorImage />
    <Title>Something went wrong...</Title>
    <Subtitle>Please try again later.</Subtitle>
  </GeneralErrorContainer>
);
