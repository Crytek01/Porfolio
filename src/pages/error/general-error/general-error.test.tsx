import { shallowRender } from '@common/utils/test/shallowRender';
import { describe, expect, it } from 'vitest';

import { GeneralErrorPage } from './general-error.page';

describe('General page component Test', () => {
  it('renders correctly', () => {
    const renderer = shallowRender(<GeneralErrorPage />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
