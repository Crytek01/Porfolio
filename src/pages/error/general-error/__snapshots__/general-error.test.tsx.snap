// Vitest Snapshot v1

exports[`General page component Test > renders correctly 1`] = `
<styled.div>
  <Styled(SvgPlugsConnectionOutlineSvg) />
  <styled.h1>
    Something went wrong...
  </styled.h1>
  <styled.p>
    Please try again later.
  </styled.p>
</styled.div>
`;
